src/esp/nav/GreedyFollower.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "esp/core/esp.h"
+#include "esp/nav/PathFinder.h"
+#include "esp/scene/SceneGraph.h"
+#include "esp/scene/SceneNode.h"
+
+namespace esp {
+namespace nav {
+
+class GreedyGeodesicFollowerImpl {
+ public:
+  enum class CODES { ERROR = -2, STOP = -1, FORWARD = 0, LEFT = 1, RIGHT = 2 };
+
+  typedef std::function<void(scene::SceneNode*)> MoveFn;
+  typedef std::tuple<vec3f, quatf> State;
+
+  GreedyGeodesicFollowerImpl(PathFinder::ptr& pathfinder,
```

Comment:
docstring with parameters (even though this is Impl, it is in the header so good to have some docs for readability and extensibility)